Feature/Retrieve Wikipedia page metadata using COINs data

rnathanday opened the pull request:
This pull request contains work started during the Oct 2014 PLOS Labs Rich Citations Hackathon.

This changeset enables retrieving COINs-encoded metadata from url's identified as Wikipedia urls.

Test cases have been included for identifying Wikipedia URLs in reference list items (`spec/classes/identifier_resolvers/wikipedia_from_reference_spec.rb`) and for fetching metadata (
`spec/classes/processors/references_info_from_wikipedia_spec.rb`).

The COINs -> Rich Citations field mapping is included in the file `app/classes/processors/references_info_from_wikipedia.rb` as:

```
'rft.type'        =>  'type',
'rft.title'       =>  'title',
'rft.date'        =>  'issued',
'rft.source'      =>  'bib_source',
'rft.aucorp'      =>  'author',
'rft.publisher'   =>  'publisher',
'rft.identifier'  =>  'URL',
'rft.language'    =>  'language',
'rft.rights'      =>  'license',
```

egh commented:
This looks sweet, thank you! I'm pretty sure the (c) statement should be your own name, though. :)

rnathanday commented:
Thanks -- hopefully it will be of some use! Could I assign copyright to PLOS?

egh commented:
copyright assignment involves a lot of legal paperwork. If you license your work using the same license we can incorporate it. Thanks!

rnathanday commented:
Not looking to generate paperwork -- copyright statements updated

egh commented:
Thanks, Nathan!

Running tests seems to be failing for me: `bundle exec rake spec SPEC="--tag ~js" RAILS_ENV=test`
Are they working for you?

The `--tag ~js` is because our javascript tests are broken. :(

rnathanday commented:
Sorry about that, Erik -- should be fixed now.

egh commented:
Thanks, Nathan! This looks great. I'm trying to process the paper 10.1371/journal.pbio.1001417
but I am getting an error:

```
failed with undefined method `sub!' for nil:NilClass
/home/egh/c/rich_citations/lib/id/url.rb:71:in `extract_date'
/home/egh/c/rich_citations/lib/id/wikipedia.rb:47:in `block in extract_from_xml'
```

It looks like your code is passing a nil into `extract_date`.

rnathanday commented:
I may have neglected to implement some methods. How did you process 10.1371/journal.pbio.1001417 ?

egh commented:
Just `rails server` and paste in 10.1371/journal.pbio.1001417 to the form on http://localhost:3000/

rnathanday commented:
Erik, I could not reproduce the nil:NilClass error locally, but I updated the code around the reported error message, and also added some tests to add coverage for the wikipedia refs in the article you referenced. All tests (rspec and manual) are passing for me locally with this commit.

egh commented:
Hi Nathan,

Thanks so much for looking into the issue.

egh commented:
oops - was still writing. :) I can't reproduce now either, so I guess it was a transient issue or your latest commit fixed it. The code is working great. Thanks for the contribution! Merging into master.
